src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1151,15 +1151,7 @@ public void Deconstruct(out int a, out int b)
 
             var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
             comp.VerifyDiagnostics(
-                // (6,33): error CS1003: Syntax error, '=>' expected
-                //         var z = ((var x, int y) = new C());
-                Diagnostic(ErrorCode.ERR_SyntaxError, "=").WithArguments("=>", "=").WithLocation(6, 33),
-                // (6,33): error CS1525: Invalid expression term '='
-                //         var z = ((var x, int y) = new C());
-                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "=").WithArguments("=").WithLocation(6, 33),
-                // (6,19): error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
-                //         var z = ((var x, int y) = new C());
-                Diagnostic(ErrorCode.ERR_TypeVarNotFound, "var").WithLocation(6, 19)
+                // this is permitted now, as it is just a deconstruction assignment expression
```

Comment:
>// this is permitted now, as it is just a deconstruction assignment expression [](start = 16, length = 78)

Please add a test that verifies that the behavior of the compiled code is correct. Test SemanticModel as well.